File: src/main/java/io/sch/historyscan/infrastructure/features/codebase/info/GitFolder.java
package io.sch.historyscan.infrastructure.features.codebase.info;

import java.io.File;

public record GitFolder(File codebaseRootFolder) {
    public File toDotGitFolder() {
        if (codebaseRootFolder.isDirectory() && !codebaseRootFolder.getName().equals(".git")) {
            return new File(codebaseRootFolder, ".git");
        } else if (codebaseRootFolder.isFile()) {
            return new File(codebaseRootFolder.getParent(), ".git");
        }
        throw new CodeBaseGitFolderNotFoundException("The codebase root folder is not a file or a directory");
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/features/codebase/info/GitFolder.kt
package io.sch.historyscan.infrastructure.features.codebase.info

import java.io.File

class GitFolder(private val codebaseRootFolder: File) {
    fun toDotGitFolder(): File {
        if (codebaseRootFolder.isDirectory && codebaseRootFolder.name != ".git") {
            return File(codebaseRootFolder, ".git")
        } else if (codebaseRootFolder.isFile) {
            return File(codebaseRootFolder.parent, ".git")
        }
        throw CodeBaseGitFolderNotFoundException("The codebase root folder is not a file or a directory")
    }
}
